orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -535,8 +535,8 @@ Status OrtLoadInternal(const PathString& checkpoint_path,
   for (auto& init : *(model_proto.mutable_graph()->mutable_initializer())) {
     ORT_ENFORCE(init.has_name(), "An initializer should have a name.");
     auto it = param_tensor_protos.find(init.name());
-    ORT_ENFORCE(it != param_tensor_protos.end(),
-                "The initializer name was not found in the checkpoint file loaded.");
+    if (it == param_tensor_protos.end())
```

Comment:
what's the reason we make the change?

Reply:
I think this was an error.

Imagine there is a checkpoint file that has only a partial list of initializers to load. The earlier code will result in a failure because the initializer in the model are not there in the checkpoint file.

We should only load the initializers in the checkpoint file on the model, without caring for the existence of all the model initializers in the checkpoint.

Reply:
Maybe we can expose a `strict` bool flag that enforces/relaxes this rule.

Reply:
Are the constant initializers stored in model proto now? and previously they is stored as non-trainable params, Am I right?

Reply:
There are three types of initializers:
1. Trainable model parameters.
2. Non trainable model parameters.
3. Constant model initializers.

Previously, 1 and 2 were in the checkpoint file, where as 3 was in model proto.

Now, only 1 is in checkpoint file, and 2 and 3 are in model proto.